service/matching/nexus_endpoint_client.go

```diff
@@ -360,3 +371,45 @@ func (m *nexusEndpointClient) resetCacheStateLocked() {
 	m.endpointsByID = make(map[string]*persistencespb.NexusEndpointEntry)
 	m.endpointsByName = make(map[string]*persistencespb.NexusEndpointEntry)
 }
+
+// notifyOwnershipChanged starts or stops a background routine which watches the Nexus endpoints table version for
+// changes. This is only expected to be called from matchingEngineImpl.notifyNexusEndpointsOwnershipChange()
+func (m *nexusEndpointClient) notifyOwnershipChanged(isOwner bool) {
+	oldIsOwner := m.isTableOwner.Swap(isOwner)
+	if isOwner && !oldIsOwner {
+		// Just acquired ownership. Start refresh loop on table version to catch any updates from previous owner.
+		backgroundCtx := headers.SetCallerInfo(
+			context.Background(),
+			headers.SystemBackgroundCallerInfo,
+		)
+		m.refreshHandle = goro.NewHandle(backgroundCtx)
```

Comment:
Was trying to avoid the extra overhead since this is only intended to be called from `watchMembership` which is single threaded, but I don't mind the extra safety

Reply:
Ah, I would accept a comment explaining that we're assuming this is single threaded.

Reply:
I see you already added the lock, that's probably better, the overhead is negligible.